src/v/archival/ntp_archiver_service.cc

```diff
@@ -1906,11 +1906,18 @@ ss::future<> ntp_archiver::garbage_collect_archive() {
         throw std::system_error(backlog.error());
     }
 
-    std::deque<std::filesystem::path> segments_to_remove;
-    std::deque<std::filesystem::path> manifests_to_remove;
+    fragmented_vector<std::filesystem::path> segments_to_remove;
+    fragmented_vector<std::filesystem::path> manifests_to_remove;
```

Comment:
I think it uses more memory in this case, because the size of the collection is usually small

Reply:
You're right. I ran a little experiment and `std::deque` uses 1KiB less for under 100 elements

Reply:
Fixed.